### ISSUE TITLE
Allow constant() to take a vector of data

### ISSUE THREAD

**Issue by jackobailey** [paul-buerkner/brms#1108]
Hi Paul,

**TL;DR: Can you let us pass a vector of data of length N to constant() to fix a parameter to a given value for each row in the data?** 

This might be a roundabout explanation, but I hope it makes sense. I'm working on a project where I'm analysing political polls over time. These take a value between 0-1 and have a known sample size. To model these data, we can use the reparameterized beta distribution where the first shape parameter α = μф and the second shape parameter β = (1 − μ)ф. Here, μ is equal to the proportion of the vote that the party got and ф the associated sample size.

brms already uses this parameterization under-the-hood, I believe. But, while it allows us to fix ф to some constant, it does not allow us to fix ф to separate constants for each row in the data. Given that these values are known in some cases, this would be a useful addition to have. In effect, it would allow uses to incorporate measurement error into models with beta likelihoods.

The reason I ask for it to be incorporated into constant() and not some new resp_ term is that there are likely also other cases where fixing values on a row-by-row basis would make sense. Thus, a general framework might be more useful and easy to implement.

Code below shows what happens when you try to pass constant() a vector:

```r

# Load packages

library(tidyverse)
library(brms)


# Set number of cases

n <- 1000


# Simulate data

dta <- 
  tibble(
    vote = inv_logit_scaled(cumsum(rnorm(n, 0, .01))),
    sample = rpois(n, 1000)
  )


# Fit model

m1 <- 
  brm(
    formula = bf(vote ~ 1),
    family = Beta(link = "logit"),
    prior = 
      prior(normal(0, 1.5), class = "Intercept") +
      prior(constant(dta$sample), class = "phi"),
    data = dta,
    chains = 1,
    cores = 1
  )

```

**Comment by jackobailey** [comment]
Note also that if you run this without including the constant prior, the phi parameter (approximately) returns the average sample size of 1000.

**Comment by paul-buerkner** [comment]
Thank you for this suggestions. You can do what you want already via 

```R
m1 <- 
  brm(
    formula = bf(vote ~ 1, phi ~ 0 + offset(sample)),
    family = Beta(link = "logit", link_phi = "identity"),
    prior = 
      prior(normal(0, 1.5), class = "Intercept"),
    data = dta,
    chains = 1,
    cores = 1
  )
```

the prior interface, specifically `constant()` is not ment to understand data as input and I currently don't see how it can well be extended in that direction.
